src/ui/public/chrome/directives/global_nav/app_switcher/app_switcher.js

```diff
@@ -72,6 +72,15 @@ uiModules
       // links don't cause full-navigation events in certain scenarios
       // so we force them when needed
       this.ensureNavigation = appSwitcherEnsureNavigation;
+
+      $scope.getTooltip = link => {
```

Comment:
I would prefer that you stuck with the controller concept here and used `this.getTooltip()` and `switcher.getTooltip()` in the view.